src/Services/RateLimitService.php

```
@@ -0,0 +1,168 @@
+                 AND attempt_time >= :cutoff",
+                ['email' => $email, 'cutoff' => $cutoff]
+            );
+        } catch (\Exception $e) {
+            error_log("Error reseteando intentos: " . $e->getMessage());
+        }
+    }
+
+    /**
+     * Obtiene el número de intentos restantes
+     */
+    public function getRemainingAttempts(string $email, string $ipAddress): int {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$ipAddress'. (undefined)

(UnusedFormalParameter)